### PR TITLE
Doxygen improvements

### PR DESCRIPTION
**Pull request by martinber**
- [x] Create a nice stylesheet for doxygen
- [X] Include md documentation from `/doc/`
    - [ ] ~Fix problems with md rendering~ Doxygen should fix some things upstream
- [ ] ~Publish everything through gh-pages or something like that~ Not until we fix md rendering

I don't know if the custom css should have only the overrides I want or if I should copy the entire doxygen stylesheet and do modifications there.

Screenshot:
Preview

References: #180

### CODE REVIEW

File: buildsystem/doxygen.cmake [hunk 1]
```diff
@@ -39,6 +39,8 @@ function(doxygen_configure)
 
 		# adapt doxygen config
 		configure_file("${BUILDSYSTEM_DIR}/templates/Doxyfile.in" "${CMAKE_BINARY_DIR}/Doxyfile" @ONLY)
+		configure_file("${BUILDSYSTEM_DIR}/templates/doxygen_custom.css.in"
```
Comment (martinber) [inline buildsystem/doxygen.cmake]: I don't really understand the buildsystem. This is the way to go?

Comment (TheJJ) [inline buildsystem/doxygen.cmake]: do you have to change values in the css at configure-time? If yes, this is the way to go  (even though you should output the file to `"${CMAKE_BINARY_DIR}/doxygen_custom.css"`). If no, you can add the css path in the `Doxyfile.in` (which is configured above and can also replace variables in that process so the css is found in the source dir).

File: buildsystem/templates/Doxyfile.in [hunk 2]
```diff
@@ -47,7 +47,7 @@ PROJECT_BRIEF          =
 # exceed 55 pixels and the maximum width should not exceed 200 pixels.
 # Doxygen will copy the logo to the output directory.
 
-PROJECT_LOGO           =
+PROJECT_LOGO           = "@CMAKE_SOURCE_DIR@/assets/logo/banner_doxygen.png"
```
Comment (TheJJ) [inline buildsystem/templates/Doxyfile.in]: does it have to be a new image? Or can we reuse the `banner.png`? I assume with a proper css we can avoid the resized redundant image.

File: buildsystem/templates/Doxyfile.in [hunk 3]
```diff
@@ -916,8 +916,8 @@ HTML_STYLESHEET        =
 # since it does not replace the standard style sheet and is therefor more
 # robust against future updates. Doxygen will copy the style sheet file to
 # the output directory.
-
-HTML_EXTRA_STYLESHEET  =
+#ASDF
+HTML_EXTRA_STYLESHEET  = "doxygen_custom.css"
```
Comment (TheJJ) [inline buildsystem/templates/Doxyfile.in]: here, you can add a variable replacement so the css file is found. (`@CMAKE_SOURCE_DIR@/buildsystem/templates/doxygen_custom.css`). Then the configuration of a `.css.in` file is not necessary.

File: buildsystem/doxygen_custom.css [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+	background-color: white;
+    padding-top: 10px;
+    padding-left: 12px;
+    padding-right: 8px;
+    padding-bottom: 1em;
+    margin: 0;
+}
+
+body {
+	background-color: #445;
```
Comment (MaanooAk) [inline buildsystem/doxygen_custom.css]: You set the background twice, line 4 and 171

Comment (martinber) [inline buildsystem/doxygen_custom.css]: You are right, also I'm mixing tabs and spaces

**Comment by martinber** [comment]
Doxygen reads markdown slightly different so we'll have issues with the existing docs. here I wrote everything I found so I remember:

------------------------------------------------

Documentation structure, I don't know how to make 'trees' to group documentation, all pages are dropped on a single place. Also I don't know what to do with those `README.md` inside some folders. Later I'll see how to do this.

------------------------------------------------

Each page uses the first h1 header for it's name, some pages have lot's of h1 headers and some have slightly unrelated titles.

------------------------------------------------

I have to find a way to include non-markdown files, like `mom.cpp` and some files on `reverse-engineering`

------------------------------------------------

That image on `README.md` isn't working. I have to look that.

------------------------------------------------

Bold tends to not work properly. The first character inside a bold section must be alphanumerical, you can't do italics with `*` if you are inside a bold section with `**`, and this thing won't work: `*NyanObject*s`

**Solution?:** Start bold with alphanumerical characters

```
**"Can I even help?"**
 v v v
**Can I even help?**
```

The alphanumerical thing happens on `contributing.md`

**Solution?:** Make italics inside bold sections with `_` instead of `*`:

```
**Please at least skip over this whole file if you want to contribute some code to *openage***!

 | | |
 v v v

**Please at least skip over this whole file if you want to contribute some code to _openage_**!
```

I found this on `contributing.md`

**Solution?**: Don't do this:

```
*NyanObject*s
```

Happened on nyan `openage-lib.md`

------------------------------------------------

For no reason we can't do bold text on the first word of a table cell.
**Solution?:** Maybe insert a non-breaking space before each bold word:
```
Technology     | Component
---------------|----------
**C++14**      | Engine core

 | | |
 v v v

Technology     | Component
---------------|----------
&nbsp;**C++14**| Engine core
```

I think this only happens currently on main `README.md`

------------------------------------------------
Links to files don't work if path starts with `/` or `./`
**Solution?:** Change it to absolute

```
[doc/media/blendomatic.md](/doc/media/blendomatic.md).
[doc/media/terrain.md](./terrain.md)

 | | |
 v v v

[doc/media/blendomatic.md](doc/media/blendomatic.md).
[doc/media/terrain.md](doc/media/terrain.md)
```

I've noted this on `terrain.md`, `blendomatic.md`, `project_structure.md`, documentation `README.md` and `status.md`

------------------------------------------------

Those nice checkboxes `[x]` don't work. Emoticons neither :cry: 

Found on `milestones.md` and nyan `conversion.md`

------------------------------------------------

For some reason these links won't work (maybe because of the .md extension?):

```
[here](https://github.com/SFTtech/nyan/blob/master/doc/nyan.md)
[nyan specification](https://github.com/SFTtech/nyan/blob/master/doc/nyan.md)
```

On `nyan.md` and nyan `examples.md`

------------------------------------------------

Doxygen automatically links to filenames. e.g. `__init.py__`, `openage::testing::TestError`, `openage/testing/testlist.py`. I've seen this happening on `testing.md` but there are more places.

**Comment by TheJJ** [comment]
Hm, that's bad. We should of course make it as compatible as possible, but the main priority should be the sanity of docs rendered with github-flavored-markdown.

The non-relative-links are pretty bad, especially since `doc/bla/bla.md` is a relative path (to the current `.md` file), `/doc/bla/bla` is repo-absolute. Maybe we should rather send patches to doxygen so the support is improved, because working around many of those issues will worsen the stuff for most other markdown renderer as I understand your findings.

**Comment by martinber** [comment]
Maybe we should generate html from /docs using something else, like pandoc.

In that case we can add a link from doxygen to static docs in the navbar. Otherwise we can link from the static docs to doxygen.

What do you think? Because I see that you have Kevin and a [website](openage.sft.mx) already so something like readthedocs might be unnecesary

**Comment by martinber** [comment]
There I rebased and squashed everything into a single commit. I left the static docs as they were (unpolished), you can see them in the "Related pages" tab, and the main `README.md` can be seen in the Doxygen main page.

AFAIK, we decided that won't continue until Doxygen support's GitHub's markdown. And another thing missing is a tree like structure for static docs.

**Pull request merged**